### PR TITLE
Rework date handling and logging

### PR DESCRIPTION
**Pull request by vrutkovs**
All dates now `datetime.datetime` so a different approach to calculating timedelta is required

### CODE REVIEW

File: zabbix_metrics_watcher.py [hunk 1]
```diff
@@ -176,24 +181,15 @@ def _send_zabbix_message(zabbix_host, osbs_master, key, value, print_command=Tru
...
-    return {k: v for k, v in completed_builds.items() if (now - v) < 3600}
-
-
-def heartbeat(zabbix_host, osbs_master):
-    while True:
-        _send_zabbix_message(zabbix_host, osbs_master,
-                             "heartbeat", int(time()), print_command=False)
-        sleep(10)
+    return {k: v for k, v in completed_builds.items()
+            if (datetime.datetime.utcnow() - v.replace(tzinfo=None)).total_seconds() < 3600}
```
Comment (twaugh) [inline zabbix_metrics_watcher.py]: Shouldn't this be `datetime.datetime.now(datetime.timezone.utc)` to get a timezone-aware object? Replacing `tzinfo` with `None` looks like it throws away the timezone information altogether.
https://docs.python.org/3.5/library/datetime.html#datetime.datetime.utcnow

File: zabbix_metrics_watcher.py [hunk 2]
```diff
@@ -52,47 +51,53 @@ def is_finished(self):
     def duration(self):
         try:
             return int(self._data['status']['duration'])/1000000000
-        except:
+        except Exception as e:
+            logger.warn('Error duration: %s' % repr(e))
```
Comment (twaugh) [inline zabbix_metrics_watcher.py]: Can use `%r` and also avoid immediate %-subst by just supplying `e` as an extra argument.

File: zabbix_metrics_watcher.py [hunk 3]
```diff
@@ -52,47 +51,53 @@ def is_finished(self):
...
-        except:
+        except Exception as e:
+            logger.warn('Error created_time: %s' % repr(e))
             return None
 
     @property
     def started_time(self):
         try:
-            timestamp = self._data['metadata']['startTimestamp']
+            timestamp = self._data['status']['startTimestamp']
```
Comment (twaugh) [inline zabbix_metrics_watcher.py:89]: Wish I'd caught this on the previous round of reviewing, sorry. :(

File: zabbix_metrics_watcher.py [hunk 4]
```diff
@@ -139,7 +144,7 @@ def send_zabbix_notification(self, zabbix_host, osbs_master, concurrent_builds):
                 logger.warn('Exit code: %s' % e.returncode)
                 logger.warn('Output: %s' % e.output)
 
-        sleep(10)
+        sleep(1)
```
Comment (twaugh) [inline zabbix_metrics_watcher.py:148]: Why is this?

**Review by twaugh — CHANGES_REQUESTED** [review]

File: zabbix_metrics_watcher.py (hunk 3 above)
Comment (vrutkovs) [reply zabbix_metrics_watcher.py:89]: IIRC startTimestamp is recorded in metadata after the build is complete (but not while the build is running), so it took some time to fix correctly

File: zabbix_metrics_watcher.py (hunk 4 above)
Comment (vrutkovs) [reply zabbix_metrics_watcher.py:148]: We're running in a single thread, so making long pauses makes metrics incorrect when a high number of builds is submitted - as we spend 10 secs to reset settings to zero after a build is complete. This is unnecessary - Zabbix can handle 1 sec delay correctly

File: zabbix_metrics_watcher.py (hunk 1 above)
Comment (vrutkovs) [reply zabbix_metrics_watcher.py]: Openshift returns date in UTC as well, but Python cannot calculate size delta if one of the objects is time aware and the other one is not. I'll try `.now()` approach though

File: zabbix_metrics_watcher.py (hunk 2 above)
Comment (vrutkovs) [reply zabbix_metrics_watcher.py]: Fixed

File: zabbix_metrics_watcher.py (hunk 1 above)
Comment (vrutkovs) [reply zabbix_metrics_watcher.py]: Fixed, now tz is used to calculate timedelta

**Review by twaugh — APPROVED** [review]
LGTM, although there are still some logging calls that use %-substitution explicitly rather than passing extra arguments for the substitutions.

**Comment by vrutkovs** [comment]
Right, I replaced all of those in the last commit

**Pull request merged**